KIPs/kip-71.md

```diff
@@ -0,0 +1,381 @@
+---
+kip: 71
+title: Dynamic gas fee pricing mechanism
+author: Woojin Lee (jared) <jared.fi@krustuniverse.com>, Junghyun Colin Kim <colin.kim@groundx.xyz>
+discussions-to: https://forum.klaytn.com/t/en-new-transaction-fee-mechanism-dynamic-gas-price-proposal/4844 
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-04-14
+---
+
+<!--You can leave these HTML comments in your merged KIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new KIPs. Note that a KIP number will be assigned by an editor. When opening a pull request to submit your KIP, please use an abbreviated title in the filename, `kip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the KIP.-->
+A dynamic gas fee pricing mechanism based on network usage, that includes fixed-per-block network fee that is partially burned.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+There is a base fee per gas in protocol, which can move up or down each block according to a formula which is a function of gas used in parent block and gas target. The algorithm results in the base fee per gas increasing when blocks are above the gas target, and decreasing when blocks are below the gas target. It repeats until base fee reachs lower bound fee or upper bound fee. The base fee per gas is partially burned. Transactions specify the maximum fee per gas they are willing to pay total. The transaction will always pay the base fee per gas of the block it was included in.
+ 
+
+## Motivation
+<!--The motivation is critical for KIPs that want to change the Klaytn protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the KIP solves. KIP submissions without sufficient motivation may be rejected outright.-->
+
+Klaytn has been following a policy of “fixed gas price“, with the gas price fixed at a certain level (e.g. 25 ston, 750 ston). This approach was adopted in the initial phase because it allows users to easily send transactions easily without having to enter the gas fee, while minimizing the volatility of transaction fees. But in the periods leading up to the gas price increase, we have seen a lot of transactions on Klaytn as well as the following problems:
```

Comment:
```suggestion
Klaytn has been following a policy of “fixed gas price“, with the gas price fixed at a certain level (e.g. 25 ston and 750 ston). This approach was adopted in the initial phase because it allows users to easily send transactions without having to predict the gas fee, while minimizing the volatility of transaction fees. But in the periods leading up to the gas price increase, we have seen a lot of transaction bursts on Klaytn with the following problems:
```